zerver/lib/soft_deactivation.py

```diff
@@ -191,6 +192,7 @@ def add_missing_messages(user_profile: UserProfile) -> None:
     recipient_ids = []
     for sub in all_stream_subs:
         stream_subscription_logs = all_stream_subscription_logs[sub["recipient__type_id"]]
+        stream = get_stream_by_id_in_realm(sub["recipient__type_id"], user_profile.realm)
```

Comment:
I guess this is incorrect as of now, since the `UserMessage` rows are not created for soft-deactivated users when a message is sent, which needs to be handled in `get_recipient_info` and then this code will not be needed since the `UserMessage` already exists and those messages will be skipped.